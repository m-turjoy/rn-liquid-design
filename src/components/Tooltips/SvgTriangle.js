import React from 'react'
import { string, number } from 'prop-types'
import Svg, { Polygon } from 'react-native-svg'
import { SvgWrapper } from './styled'

const adjustTrianglePoints = side => ((side === 'bottom-left' || side === 'bottom-right') &&
      '11,20 20,10 29,20') ||
    ((side === 'top-left' || side === 'top-right') && '11,10 20,20 29,10') ||
    ((side === 'left-top' || side === 'left-bottom') && '10,11 10,29 20,20') ||
    ((side === 'right-top' || side === 'right-bottom') && '30,11 30,29 20,20')

const adjustTopPosition = (side, positionY, iconHeight) => (
  ((side === 'bottom-left' || side === 'bottom-right') && positionY + iconHeight + 3) ||
  ((side === 'top-left' || side === 'top-right') && positionY - 11) ||
  ((side === 'left-top' || side === 'right-top' || side === 'left-bottom'
    || side === 'right-bottom') &&
  positionY + ((iconHeight - 17) / 2))
)

const adjustLeftPosition = (side, positionX, iconWidth) => (
  ((side === 'bottom-left' || side === 'bottom-right' ||
    side === 'top-left' || side === 'top-right') && ((positionX + (iconWidth / 2)) - 8)) ||
  ((side === 'left-top' || side === 'left-bottom') && positionX - 11) ||
  ((side === 'right-top' || side === 'right-bottom') && positionX + iconWidth + 3)
)
const SvgTriangle = ({
  modalRenderSide,
  modalBackgroundColor,
  positionX,
  positionY,
  iconHeight,
  iconWidth
}) => (
  <SvgWrapper
    position='absolute'
    top={adjustTopPosition(modalRenderSide, positionY, iconHeight)}
    left={adjustLeftPosition(modalRenderSide, positionX, iconWidth)}
    justifyContent='flex-start'
    alignItems='flex-start'
    elevation={15}
  >
    <Svg
      width={
        ((modalRenderSide === 'bottom-right' || modalRenderSide === 'bottom-left' ||
          modalRenderSide === 'top-left' || modalRenderSide === 'top-right') && 16) ||
       ((modalRenderSide === 'right-top' || modalRenderSide === 'left-top' ||
          modalRenderSide === 'right-bottom' || modalRenderSide === 'left-bottom') && 8)
      }
      height={
        ((modalRenderSide === 'bottom-right' || modalRenderSide === 'bottom-left' ||
          modalRenderSide === 'top-left' || modalRenderSide === 'top-right') && 8) ||
        ((modalRenderSide === 'right-top' || modalRenderSide === 'left-top' ||
          modalRenderSide === 'right-bottom' || modalRenderSide === 'left-bottom') && 17)
      }
    >
      <Polygon
        y={
          ((modalRenderSide === 'bottom-right' || modalRenderSide === 'bottom-left') && -7) ||
          ((modalRenderSide === 'top-left' || modalRenderSide === 'top-right') && -15) ||
          ((modalRenderSide === 'right-top' || modalRenderSide === 'left-top' ||
            modalRenderSide === 'right-bottom' || modalRenderSide === 'left-bottom') && -11)
        }
        x={
          ((modalRenderSide === 'bottom-right' || modalRenderSide === 'bottom-left' ||
          modalRenderSide === 'top-left' || modalRenderSide === 'top-right') && -12) ||
          ((modalRenderSide === 'right-top' || modalRenderSide === 'right-bottom') && -17) ||
          ((modalRenderSide === 'left-top' || modalRenderSide === 'left-bottom') && -15)
        }
        stroke={modalBackgroundColor}
        strokeWidth={5}
        strokeLinecap='round'
        strokeLinejoin='round'
        points={adjustTrianglePoints(modalRenderSide)}
        fill={modalBackgroundColor}
      />
    </Svg>
  </SvgWrapper>
)

SvgTriangle.propTypes = {
  modalRenderSide: string.isRequired,
  modalBackgroundColor: string.isRequired,
  positionX: number.isRequired,
  positionY: number.isRequired,
  iconWidth: number.isRequired,
  iconHeight: number.isRequired
}

export default SvgTriangle
