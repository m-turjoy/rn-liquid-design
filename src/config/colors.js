const colors = {
  transparent: 'transparent',
  richPurpleDarkest: '#281949',
  richPurpleLightest: '#b9add3',
  vibrantMagentaDarkest: '#761e4b',
  richPurpleDarker: '#382366',
  vibrantMagentaDarker: '#a52a69',
  richPurpleDefault: '#503291',
  richPurpleDark: '#482d83',
  richPurpleLighter: '#9684bd',
  richPurpleLight: '#735ba7',
  microsoftPurple: '#5457ae',
  vibrantMagentaDark: '#d43687',
  vibrantMagentaLight: '#ef63ab',
  vibrantMagentaLighter: '#f38ac0',
  vibrantMagentaDefault: '#eb3c96',
  richBlueDarkest: '#083558',
  vibrantMagentaLightest: '#f7b1d5',
  richBlueDark: '#0e5f9e',
  richBlueDarker: '#0b4a7b',
  richBlueDefault: '#0f69af',
  richBlueLight: '#3f87bf',
  richBlueLighter: '#6fa5cf',
  richBlueLightest: '#9fc3df',
  facebookBlue: '#3b5998',
  twitterBlue: '#1da1f2',
  linkedInBlue: '#0274b3',
  flickrBlue: '#025fdf',
  mailBlue: '#5877ff',
  salesforceBlue: '#00a1e0',
  skypeBlue: '#4aaee6',
  vibrantCyanDarker: '#208590',
  vibrantCyanDarkest: '#175f67',
  vibrantCyanDefault: '#2dbecd',
  vibrantCyanLight: '#57cbd7',
  vibrantYellowDarkest: '#806419',
  vibrantCyanLightest: '#abe5eb',
  vibrantYellowDarker: '#ddac28',
  vibrantYellowDark: '#f3be2f',
  vibrantCyanLighter: '#81d8e1',
  vibrantYellowDefault: '#ffc832',
  vibrantYellowLight: '#ffd35b',
  vibrantYellowLighter: '#ffde84',
  vibrantYellowLightest: '#ffe9ad',
  snapchatYellow: '#fffc00',
  vibrantGreenDarkest: '#536728',
  vibrantGreenDarker: '#749038',
  vibrantGreenDark: '#95b948',
  vibrantGreenDefault: '#a5cd50',
  vibrantGreenLight: '#b7d773',
  vibrantGreenLighter: '#c9e196',
  vibrantGreenLightest: '#dbebb9',
  richRedDark: '#cf1b48',
  richRedDefault: '#e61e50',
  richRedLight: '#eb4b73',
  richRedDarkest: '#730f28',
  richRedLighter: '#f07896',
  instagramRed: '#ca486f',
  richBlackDefault: '#1b1b25',
  snapchatBlack: '#010101',
  richRedLightest: '#f5a5b9',
  richBlackLight: '#494953',
  richBlackLighter: '#6a6a73',
  richBlackLightest: '#93939d',
  sensitiveGreyDarkest: '#e9e9ed',
  sensitiveGreyDarker: '#efeff1',
  sensitiveGreyDark: '#f3f3f7',
  sensitiveGreyDefault: '#f8f8fc',
  sensitiveGreyLight: '#f8f8fc',
  sensitiveGreyLighter: '#ffffff',
  sensitiveGreyLightest: '#ffffff',
  sensitiveGreyLabel: '#e9e9ed',
  richRedDarker: '#a11538',
  white: '#ffffff',
  richGreenDarkest: '#094d2f',
  richGreenDarker: '#0c5d39',
  richGreenDark: '#0d6c42',
  richGreenDefault: '#149b5f',
  richGreenLight: '#5ab98f',
  richGreenLighter: '#72c39f',
  richGreenLightest: '#89cdaf',
  sensitiveGreen: '#b4dc96',
  xingGreen: '#005b5f',
  slackGreen: '#3eb991',
  sensitiveYellow: '#ffdcb9',
  sensitiveBlue: '#96d7d2',
  sensitivePink: '#e1c3cd',
  vibrantCyanDark: '#29abb9'
}

export default colors
