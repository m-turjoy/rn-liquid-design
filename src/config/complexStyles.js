import { complexStyle } from 'styled-system';

const shadowStyle = complexStyle({
  prop: 'shadowStyle',
  key: 'shadows',
});
export { shadowStyle };
