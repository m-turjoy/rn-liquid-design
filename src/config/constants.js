import { Dimensions } from 'react-native';

const constants = {
  defaultMargin: 10,
  drawerWidth: 245,
  // / /\ TO DELETE LATER
  deviceWidth: Dimensions.get('window').width,
  deviceHeight: Dimensions.get('window').height,
};
export default constants;
