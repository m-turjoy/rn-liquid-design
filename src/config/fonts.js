const fonts = {
  Black: 'Lato-Black',
  Bold: 'Lato-Bold',
  Regular: 'Lato-Regular',
  Merck: 'MerckWeb-Regular',
  IcoMoon: 'icomoon'
}

export default fonts
