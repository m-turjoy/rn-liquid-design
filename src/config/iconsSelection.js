const iconsSelection = (iconName) => {
  switch (iconName) {
    case 'attention':
      return '\ue9a0';
    case 'information':
      return '\ue9ab';
    case 'merckLogo':
      return '\ue9c7';
    case 'facebook':
      return '\ue9ba';
    case 'flickr':
      return '\ue9bb';
    case 'instagram':
      return '\ue9bc';
    case 'linked-in':
      return '\ue9bd';
    case 'mail':
      return '\ue9be';
    case 'salesforce':
      return '\ue9bf';
    case 'skype':
      return '\ue9c0';
    case 'slack':
      return '\ue9c1';
    case 'twitter':
      return '\ue9c4';
    case 'xing':
      return '\ue9c5';
    case 'checkboxFilled':
      return '\ue999';
    case 'checkboxEmpty':
      return '\ue9a3';
    case 'progressBarIndicator':
      return '\ue900';
    case 'tooltipEmpty':
      return '\ue903';
    case 'arrowChevron':
      return '\ue904';
    case 'progressBarCheck':
      return '\ue905';
    case 'progressBarComing':
      return '\ue906';
    case 'progressBarDisabled':
      return '\ue907';
    case 'radioBtnActive':
      return '\ue908';
    case 'radioBtnEmpty':
      return '\ue909';
    case 'radioBtnFilled':
      return '\ue90a';
    case 'radioBtnHover':
      return '\ue90b';
    case 'calendar':
      return '\ue90c';
    case 'listview-filled':
      return '\ue90d';
    case 'listview-line':
      return '\ue90e';
    case 'alarm-fill':
      return '\ue90f';
    case 'alarm-line':
      return '\ue910';
    case 'atoms-fill':
      return '\ue911';
    case 'atoms-line':
      return '\ue912';
    case 'basket-filled':
      return '\ue913';
    case 'basket-line':
      return '\ue914';
    case 'bin-filled':
      return '\ue915';
    case 'bin-line':
      return '\ue916';
    case 'biotech-filled':
      return '\ue917';
    case 'biotech-line':
      return '\ue918';
    case 'bitcoin-filled':
      return '\ue919';
    case 'bitcoin-line':
      return '\ue91a';
    case 'bottle-filled':
      return '\ue91b';
    case 'bottle-line':
      return '\ue91c';
    case 'bulletList-filled':
      return '\ue91d';
    case 'bulletList-line':
      return '\ue91e';
    case 'camera-filled':
      return '\ue91f';
    case 'camera-line':
      return '\ue920';
    case 'chatboxRound-filled':
      return '\ue921';
    case 'chatboxRound-line':
      return '\ue922';
    case 'chatboxSquare-filled':
      return '\ue923';
    case 'chatboxSquare-line':
      return '\ue924';
    case 'chats-filled':
      return '\ue925';
    case 'chats-line':
      return '\ue926';
    case 'clip-filled':
      return '\ue927';
    case 'clip-line':
      return '\ue928';
    case 'cloudDownload-filled':
      return '\ue929';
    case 'cloudDownload-line':
      return '\ue92a';
    case 'cloudUpload-filled':
      return '\ue92b';
    case 'cloudUpload-line':
      return '\ue92c';
    case 'costCenter-filled':
      return '\ue92d';
    case 'costCenter-line':
      return '\ue92e';
    case 'coupon-filled':
      return '\ue92f';
    case 'coupon-line':
      return '\ue930';
    case 'dashboard-filled':
      return '\ue931';
    case 'dashboard-line':
      return '\ue932';
    case 'document-filled':
      return '\ue933';
    case 'document-line':
      return '\ue934';
    case 'dollar-filled':
      return '\ue935';
    case 'dollar-line':
      return '\ue936';
    case 'dollarCoins-filled':
      return '\ue937';
    case 'dollarCoins-line':
      return '\ue938';
    case 'dollarSign-filled':
      return '\ue939';
    case 'dollarSign-line':
      return '\ue93a';
    case 'education-filled':
      return '\ue93b';
    case 'education-line':
      return '\ue93c';
    case 'energy-filled':
      return '\ue93d';
    case 'energy-line':
      return '\ue93e';
    case 'euro-filled':
      return '\ue93f';
    case 'euro-line':
      return '\ue940';
    case 'filter-filled':
      return '\ue941';
    case 'filter-line':
      return '\ue942';
    case 'finance-filled':
      return '\ue943';
    case 'finance-line':
      return '\ue944';
    case 'gridView-filled':
      return '\ue945';
    case 'gridView-line':
      return '\ue946';
    case 'growth-filled':
      return '\ue947';
    case 'growth-line':
      return '\ue948';
    case 'headphone-filled':
      return '\ue949';
    case 'headphone-line':
      return '\ue94a';
    case 'heart-filled':
      return '\ue94b';
    case 'heart-line':
      return '\ue94c';
    case 'historicalData-filled':
      return '\ue94d';
    case 'historicalData-line':
      return '\ue94e';
    case 'home-filled':
      return '\ue94f';
    case 'home-line':
      return '\ue950';
    case 'honeycomb-filled':
      return '\ue951';
    case 'honeycomb-line':
      return '\ue952';
    case 'income-filled':
      return '\ue953';
    case 'income-line':
      return '\ue954';
    case 'keys-filled':
      return '\ue955';
    case 'keys-line':
      return '\ue956';
    case 'language-filled':
      return '\ue957';
    case 'language-line':
      return '\ue958';
    case 'laptopMobile-filled':
      return '\ue959';
    case 'laptopMobile-line':
      return '\ue95a';
    case 'layer-filled':
      return '\ue95b';
    case 'layer-line':
      return '\ue95c';
    case 'location-filled':
      return '\ue95d';
    case 'location-line':
      return '\ue95e';
    case 'lock-filled':
      return '\ue95f';
    case 'lock-line':
      return '\ue960';
    case 'logistic-filled':
      return '\ue961';
    case 'logistic-line':
      return '\ue962';
    case 'logisticUp-filled':
      return '\ue963';
    case 'logisticUp-line':
      return '\ue964';
    case 'message-filled':
      return '\ue965';
    case 'message-line':
      return '\ue966';
    case 'mic-filled':
      return '\ue967';
    case 'mic-line':
      return '\ue968';
    case 'microscope-filled':
      return '\ue969';
    case 'microscope-line':
      return '\ue96a';
    case 'money-filled':
      return '\ue96b';
    case 'money-line':
      return '\ue96c';
    case 'movie-filled':
      return '\ue96d';
    case 'movie-line':
      return '\ue96e';
    case 'pencil-filled':
      return '\ue96f';
    case 'pencil-line':
      return '\ue970';
    case 'people1-filled':
      return '\ue971';
    case 'people1-line':
      return '\ue972';
    case 'people2-filled':
      return '\ue973';
    case 'people2-line':
      return '\ue974';
    case 'person-filled':
      return '\ue975';
    case 'person-line':
      return '\ue976';
    case 'phone-filled':
      return '\ue977';
    case 'phone-line':
      return '\ue978';
    case 'pipette-filled':
      return '\ue979';
    case 'pipette-line':
      return '\ue97a';
    case 'plane-filled':
      return '\ue97b';
    case 'plane-line':
      return '\ue97c';
    case 'posting-filled':
      return '\ue97d';
    case 'posting-line':
      return '\ue97e';
    case 'pound-filled':
      return '\ue97f';
    case 'pound-line':
      return '\ue980';
    case 'puzzle-filled':
      return '\ue981';
    case 'puzzle-line':
      return '\ue982';
    case 'reaction-filled':
      return '\ue983';
    case 'reaction-line':
      return '\ue984';
    case 'repost-filled':
      return '\ue985';
    case 'repost-line':
      return '\ue986';
    case 'roundFlask-filled':
      return '\ue987';
    case 'roundFlask-line':
      return '\ue988';
    case 'settings-filled':
      return '\ue989';
    case 'settings-line':
      return '\ue98a';
    case 'share-filled':
      return '\ue98b';
    case 'share-line':
      return '\ue98c';
    case 'speaker-filled':
      return '\ue98d';
    case 'speaker-line':
      return '\ue98e';
    case 'testTube-filled':
      return '\ue98f';
    case 'testTube-line':
      return '\ue990';
    case 'tools-filled':
      return '\ue991';
    case 'tools-line':
      return '\ue992';
    case 'triangularFlask-filled':
      return '\ue993';
    case 'triangularFlask-line':
      return '\ue994';
    case 'truck-filled':
      return '\ue995';
    case 'truck-line':
      return '\ue996';
    case 'watch-filled':
      return '\ue997';
    case 'watch-line':
      return '\ue998';
    case 'tooltipFilled':
      return '\ue99a';
    case 'add':
      return '\ue99b';
    case 'arrowDown':
      return '\ue99c';
    case 'arrowLeft':
      return '\ue99d';
    case 'arrowRight':
      return '\ue99e';
    case 'arrowUp':
      return '\ue99f';
    case 'attention1':
      return '\ue9c8';
    case 'bell':
      return '\ue9a1';
    case 'checkArrow':
      return '\ue9a2';
    case 'closingX':
      return '\ue9a4';
    case 'dot':
      return '\ue9a5';
    case 'doubleArrowLeft':
      return '\ue9a6';
    case 'doubleArrowRight':
      return '\ue9a7';
    case 'export':
      return '\ue9a8';
    case 'favorite':
      return '\ue9a9';
    case 'filter':
      return '\ue9aa';
    case 'information1':
      return '\ue9c6';
    case 'mediaAudio':
      return '\ue9ac';
    case 'mediaFastForward':
      return '\ue9ad';
    case 'mediaPause':
      return '\ue9ae';
    case 'mediaPlay':
      return '\ue9af';
    case 'mediaRewind':
      return '\ue9b0';
    case 'mediaStop':
      return '\ue9b1';
    case 'minus':
      return '\ue9b2';
    case 'options':
      return '\ue9b3';
    case 'placeholder':
      return '\ue9b4';
    case 'plus':
      return '\ue9b5';
    case 'search':
      return '\ue9b6';
    case 'settings':
      return '\ue9b7';
    case 'star':
      return '\ue9b8';
    case 'upload':
      return '\ue9b9';
    default:
      return '\ue9b4';
  }
};

export default iconsSelection;
