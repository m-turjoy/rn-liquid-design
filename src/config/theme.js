import { Platform } from 'react-native'
import colors from './colors'
import fonts from './fonts'

const theme = {
  fontWeights: [100, 200, 300, 400, 500, 600, 700, 800, 900],
  widths: [50, 100, 130],
  heights: {
    Default: 40,
    Big: 50
  },
  fontSizes: [8, 10, 12, 14, 16, 18, 20, 24, 36],
  space: [0, 1, 2, 5, 10, 15, 20, 24, 50, 8, 7, 4, 9, 6],
  fonts,
  shadows: {
    default: {},
    active: {
      shadowColor: colors.richBlackDefault,
      shadowOpacity: 0.3,
      shadowOffset: '0px 30px',
      shadowRadius: 20,
      elevation: 40
    },
    stacked: {
      shadowColor: colors.richBlackDefault,
      shadowOpacity: 0.3,
      shadowOffset: '0px 30px',
      shadowRadius: 20,
      elevation: 25
    }
  },
  stacked: {
    stackedFirst: {
      position: 'absolute',
      zIndex: 10,
      borderRightWidth: 1,
      borderBottomWidth: 1,
      borderColor: colors.sensitiveGreyDark,
      ...(Platform.OS === 'android'
        ? {
          borderLeftWidth: 1,
          borderRightColor: colors.sensitiveGreyDark,
          borderLeftColor: colors.white,
          borderBottomColor: colors.sensitiveGreyDark,
          borderBottomLeftRadius: 10
        }
        : {})
    },
    stackedSecond: {
      position: 'absolute',
      zIndex: 9,
      top: 3,
      left: 3,
      borderRightWidth: 1,
      borderBottomWidth: 1,
      borderColor: colors.sensitiveGreyDark,
      ...(Platform.OS === 'android'
        ? {
          borderLeftWidth: 1,
          borderRightColor: colors.sensitiveGreyDark,
          borderLeftColor: colors.white,
          borderBottomColor: colors.sensitiveGreyDark,
          borderBottomLeftRadius: 10
        }
        : {})
    },
    stackedThird: {
      zIndex: 8,
      top: 6,
      left: 6,
      borderRightWidth: 1,
      borderBottomWidth: 1,
      borderColor: colors.sensitiveGreyDark,
      ...(Platform.OS === 'android'
        ? {
          borderLeftWidth: 1,
          borderRightColor: colors.sensitiveGreyDark,
          borderLeftColor: colors.white,
          borderBottomColor: colors.sensitiveGreyDark,
          borderBottomLeftRadius: 10
        }
        : {})
    }
  }
}

const themeColors = {
  richPurple: {
    lightest: colors.richPurpleLightest,
    lighter: colors.richPurpleLighter,
    light: colors.richPurpleLight,
    base: colors.richPurpleDefault,
    dark: colors.richPurpleDark,
    darker: colors.richPurpleDarker,
    darkest: colors.richPurpleDarkest
  },
  vibrantMagenta: {
    lightest: colors.vibrantMagentaLightest,
    lighter: colors.vibrantMagentaLighter,
    light: colors.vibrantMagentaLight,
    base: colors.vibrantMagentaDefault,
    dark: colors.vibrantMagentaDark,
    darker: colors.vibrantMagentaDarker,
    darkest: colors.vibrantMagentaDarkest
  },
  richBlue: {
    lightest: colors.richBlueLightest,
    lighter: colors.richBlueLighter,
    light: colors.richBlueLight,
    base: colors.richBlueDefault,
    dark: colors.richBlueDark,
    darker: colors.richBlueDarker,
    darkest: colors.richBlueDarkest
  },
  vibrantCyan: {
    lightest: colors.vibrantCyanLightest,
    lighter: colors.vibrantCyanLighter,
    light: colors.vibrantCyanLight,
    base: colors.vibrantCyanDefault,
    dark: colors.vibrantCyanDark,
    darker: colors.vibrantCyanDarker,
    darkest: colors.vibrantCyanDarkest
  },
  vibrantYellow: {
    lightest: colors.vibrantYellowLightest,
    lighter: colors.vibrantYellowLighter,
    light: colors.vibrantYellowLight,
    base: colors.vibrantYellowDefault,
    dark: colors.vibrantYellowDark,
    darker: colors.vibrantYellowDarker,
    darkest: colors.vibrantYellowDarkest
  },
  vibrantGreen: {
    lightest: colors.vibrantgGreenLightest,
    lighter: colors.vibrantgGreenLighter,
    light: colors.vibrantgGreenLight,
    base: colors.vibrantgGreenDefault,
    dark: colors.vibrantgGreenDark,
    darker: colors.vibrantgGreenDarker,
    darkest: colors.vibrantgGreenDarkest
  },
  richRed: {
    lightest: colors.richRedLightest,
    lighter: colors.richRedLighter,
    light: colors.richRedLight,
    base: colors.richRedDefault,
    dark: colors.richRedDark,
    darker: colors.richRedDarker,
    darkest: colors.richRedDarkest
  },
  richBlack: {
    lightest: colors.richBlackCyanLightest,
    lighter: colors.richBlackCyanLighter,
    light: colors.richBlackCyanLight,
    base: colors.richBlackCyanDefault,
    dark: colors.richBlackCyanDark,
    darker: colors.richBlackCyanDarker,
    darkest: colors.richBlackCyanDarkest
  },
  sensitiveGrey: {
    lightest: colors.sensitiveGreyLightest,
    lighter: colors.sensitiveGreyLighter,
    light: colors.sensitiveGreyLight,
    base: colors.sensitiveGreyDefault,
    dark: colors.sensitiveGreyDark,
    darker: colors.sensitiveGreyDarker,
    darkest: colors.sensitiveGreyDarkest
  },
  white: {
    base: colors.white
  },
  richGreen: {
    lightest: colors.richGreenLightest,
    lighter: colors.richGreenLighter,
    light: colors.richGreenLight,
    base: colors.richGreenDefault,
    dark: colors.richGreenDark,
    darker: colors.richGreenDarker,
    darkest: colors.richGreenDarkest
  },
  black: {
    base: colors.black
  }
}

const getTheme = ({ primary, secondary }, props = {}) => ({
  colors: {
    primary: {
      lightest: colors[`${primary}Lightest`],
      light: colors[`${primary}Light`],
      base: colors[`${primary}Default`],
      dark: colors[`${primary}Dark`],
      darker: colors[`${primary}Darker`]
    },
    secondary: {
      lightest: colors[`${secondary}Lightest`],
      light: colors[`${secondary}Light`],
      base: colors[`${secondary}Default`],
      dark: colors[`${secondary}Dark`],
      darker: colors[`${secondary}Darker`]
    },
    ...themeColors
  },
  ...theme,
  ...props
})

const themes = {
  vibrantCyan: getTheme({
    primary: 'vibrantCyan',
    secondary: 'vibrantYellow'
  }),
  richBlue: getTheme({
    primary: 'richBlue',
    secondary: 'vibrantYellow'
  }),
  richPurple: getTheme(
    {
      primary: 'richPurple',
      secondary: 'vibrantCyan'
    }
  ),
  vibrantMagenta: getTheme(
    {
      primary: 'vibrantMagenta',
      secondary: 'richPurple'
    }
  )
}

export const defaultThemeName = 'vibrantCyan'
export const defaultTheme = themes[defaultThemeName]

export const getThemeObject = (theme) => {
  let themeObject = null
  if (theme && typeof theme === 'object') {
    themeObject = { ...themes[defaultThemeName] }
    themeObject.colors = { ...theme, ...themeColors }
  }
  if (typeof theme === 'string') {
    themeObject = themes[theme]
    if (!themeObject) {
      const themesString = Object.keys(themes).join(', ')
      throw new Error(
        `Invalid theme name specified. Choose one from following: ${themesString}.`
      )
    }
  }

  return themeObject
}

export default { theme, themes }
