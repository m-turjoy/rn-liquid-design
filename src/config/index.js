import colors from './colors'
import fonts from './fonts'
import constants from './constants'
import theme from './theme'

export { colors, constants, theme, fonts }
