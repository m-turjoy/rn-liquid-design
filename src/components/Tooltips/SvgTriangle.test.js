import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SvgTriangle from './SvgTriangle';
import { colors } from '../../config';

configure({ adapter: new Adapter() });
describe('SvgTriangle component', () => {
  describe('Renders correctly', () => {
    test('it renders default SvgTriangle', () => {
      const tree = shallow(
        <SvgTriangle
          modalRenderSide="bottom-left"
          modalBackgroundColor={colors.white}
          positionX={0}
          positionY={0}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
