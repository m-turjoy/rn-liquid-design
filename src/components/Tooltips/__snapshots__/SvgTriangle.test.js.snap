// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgTriangle component Renders correctly it renders default SvgTriangle 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SvgTriangle
    modalBackgroundColor="#ffffff"
    modalRenderSide="bottom-left"
    positionX={0}
    positionY={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "flex-start",
      "children": <Svg
        height={8}
        preserveAspectRatio="xMidYMid meet"
        width={16}
      >
        <Polygon
          fill="#ffffff"
          points="11,20 20,10 29,20"
          stroke="#ffffff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={5}
          x={-12}
          y={-7}
        />
      </Svg>,
      "elevation": 15,
      "justifyContent": "flex-start",
      "left": false,
      "position": "absolute",
      "top": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Polygon
          fill="#ffffff"
          points="11,20 20,10 29,20"
          stroke="#ffffff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={5}
          x={-12}
          y={-7}
        />,
        "height": 8,
        "preserveAspectRatio": "xMidYMid meet",
        "width": 16,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fill": "#ffffff",
          "points": "11,20 20,10 29,20",
          "stroke": "#ffffff",
          "strokeLinecap": "round",
          "strokeLinejoin": "round",
          "strokeWidth": 5,
          "x": -12,
          "y": -7,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "flex-start",
        "children": <Svg
          height={8}
          preserveAspectRatio="xMidYMid meet"
          width={16}
        >
          <Polygon
            fill="#ffffff"
            points="11,20 20,10 29,20"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={5}
            x={-12}
            y={-7}
          />
        </Svg>,
        "elevation": 15,
        "justifyContent": "flex-start",
        "left": false,
        "position": "absolute",
        "top": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Polygon
            fill="#ffffff"
            points="11,20 20,10 29,20"
            stroke="#ffffff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={5}
            x={-12}
            y={-7}
          />,
          "height": 8,
          "preserveAspectRatio": "xMidYMid meet",
          "width": 16,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fill": "#ffffff",
            "points": "11,20 20,10 29,20",
            "stroke": "#ffffff",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 5,
            "x": -12,
            "y": -7,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
